src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        final List<Machine> emptyMachineList = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        }
+        if (type == Truck.class) {
+            return new TruckProducer().get();
+        }
+        if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        return emptyMachineList;
+    }
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method. This would improve code readability and maintainability. For instance, you could declare a variable for the producer and then return the result of the producer's `get()` method.